intelmq/bots/parsers/misp/parser.py

```diff
@@ -60,60 +60,62 @@ class MISPParserBot(Bot):
-                    event.add('raw', json.dumps(misp_event, sort_keys=True))
-                    event.add(self.MISP_TYPE_MAPPING[type_], value)
-                    event.add('comment', comment)
-                    event.add('event_description.text', category)
-                    event.add('event_description.url', misp_event_url)
-                    event.add('malware.name', malware_variant)
-                    event.add('classification.type', classifier)
-                    event.add('time.source', '{} UTC'.format(
-                              datetime.utcfromtimestamp(float(timestamp))))
-                    self.send_message(event)
+        misp_event = json.loads(raw_report)
+
+        # Set the classifier based on the ecsirt tag
+        classifier = None
+        if misp_event.get('Tag'):
+            for tag in misp_event['Tag']:
+                if tag['name'] in self.MISP_TAXONOMY_MAPPING:
+                    classifier = self.MISP_TAXONOMY_MAPPING[tag['name']]
+                    break
+
+        # get the attributes from the event
+        event_attributes = misp_event['Attribute']
+
+        # payload type - get malware variant for the event
+        malware_variant = None
+        for attribute in event_attributes:
+            if attribute['category'] == 'Payload type':
+                value = attribute['value'].lower()
+                # TODO: use misp galaxies
+                if value and harmonization.MalwareName.is_valid(value):
+                    malware_variant = value
+
+        # MISP event URL
+        url_path = 'event/view/{}'.format(misp_event['id'])
+        misp_event_url = urljoin(report['feed.url'], url_path)
+
+        # Process MISP event attributes as separate IntelMQ events
+        for attribute in event_attributes:
+
+            # get details of attribute
+            value = attribute['value']
+            uuid = attribute['uuid']
+            comment = attribute['comment']
+            timestamp = attribute['timestamp']
+            category = attribute['category']
+            type_ = attribute['type']
+
+            # create intelmq events based on the category
+            if (category in self.SUPPORTED_MISP_CATEGORIES and
+                    type_ in self.MISP_TYPE_MAPPING):
+
+                # Create and send the intelmq event
+                event = Event(report)
+                # FIXME: Send the whole MISP event with each attribute?
+                event.add('raw', json.dumps(misp_event, sort_keys=True))
+                event.add(self.MISP_TYPE_MAPPING[type_], value)
+                event.add('misp_uuid', misp_event['uuid'])
+                event.add('misp_attribute_uuid', uuid)
+                event.add('comment', comment)
+                event.add('event_description.text', category)
+                event.add('event_description.url', misp_event_url)
+                event.add('malware.name', malware_variant)
+                event.add('classification.type', classifier)
```

Comment:
In the current code, the only possible classifiers you will have are the values in the dict `MISP_TAXONOMY_MAPPING` (based on the ecsirt taxonomy), but it will only work if the MISP events are tagged with the ecsirt taxonomy, any other taxonomy will be ignored.

What do you think of a more generic approach that would use the taxonomies in https://github.com/MISP/misp-taxonomies, and import all those taxonomies by default in intelmq?